### PR TITLE
oc_clock_time_monotonic: use monotonic time from the boot time

### PR DESCRIPTION
**Pull request by jkralik**

### CODE REVIEW

File: port/esp32/adapter/src/clock.c [hunk 1]
```diff
@@ -43,12 +45,8 @@ oc_clock_time(void)
 oc_clock_time_t
 oc_clock_time_monotonic(void)
 {
-  struct timespec t;
-  if (clock_gettime(CLOCK_MONOTONIC, &t) == -1) {
-    return -1;
-  }
-  return (oc_clock_time_t)t.tv_sec * OC_CLOCK_SECOND +
-         (oc_clock_time_t)ceil((double)t.tv_nsec / (1.e09 / OC_CLOCK_SECOND));
+  // return esp_timer_get_time returns microseconds from the boot time
```
Comment (Danielius1922) [inline port/esp32/adapter/src/clock.c:48]: `OC_CLOCK_SECOND` might not be microseconds, so you need to do conversion to oc_clock_time_t

File: port/windows/clock.c [hunk 2]
```diff
@@ -60,27 +54,9 @@ oc_clock_time(void)
...
-  // is sufficiently large.
-  const LONGLONG whole =
-    (qtime.QuadPart / g_query_perm_frequency) * OC_CLOCK_SECOND;
-  const LONGLONG part =
-    (LONGLONG)((qtime.QuadPart % g_query_perm_frequency) *
-               ((double)OC_CLOCK_SECOND / g_query_perm_frequency));
-  return whole + part;
+  // The return value is the number of milliseconds that have elapsed since the
+  // system was started.
+  return ((oc_clock_time_t)GetTickCount64()) * 1000;
```
Comment (Danielius1922) [inline port/windows/clock.c:59]: Why not use the more precise `QueryPerformanceCounter`. Also you need to do conversion to oc_clock_time_t

Comment (jkralik) [reply port/windows/clock.c:59]: Note that the QueryPerformanceCounter() function is hardware-dependent and may not be available on all systems. However, it is generally available on modern Windows systems and provides a high-resolution timer that is suitable for many timing applications.

File: port/esp32/adapter/src/clock.c (hunk 1 above)
Comment (jkralik) [reply port/esp32/adapter/src/clock.c:48]: fixed

**Comment by jkralik** [comment]
~format

**Pull request merged**